module/Dashboard/config/autoload/dao/NewRelicDao.config.php

```diff
@@ -31,10 +37,14 @@
                 'https://api.newrelic.com/v2/servers/:serverId:/metrics/data.json'
                 . '?names[]=System/Memory/Used/bytes&values[]=average_response_time'
                 . '&values[]=average_exclusive_time&from=:beginDateTime:&end=:endDateTime:&summarize=true',
-            'fetchMemoryAvgForNumberWidget' =>
+            NewRelicAddresses::AVG_HOST_MEMORY =>
                 'https://api.newrelic.com/v2/applications/:appId:/metrics/data.json'
                 . '?names[]=Memory/Physical&values[]=used_mb_by_host&summarize=true'
```

Comment:
See [this](https://docs.newrelic.com/docs/apis/rest-api-v2/application-examples-v2/get-host-memory-used-application#api-call).

`used_mb_by_host` is a proper field when you specify a host in URL:
`api.newrelic.com/v2/applications/${APP_ID}/hosts/${HOST_ID}/metrics/data.json?names[]=Memory/Physical&values[]=used_mb_by_host&summarize=true`

I'm not sure why, but we use it without specifying the host. It probably matches total consumption when you use one host for the application, but it's wrong if you use multiple. It's not documented how it works if you skip the host part, but it works. I've played around and it seems such construct returns **average memory usage for _the first host_** - totally not what you'd want.

So I've added a `TOTAL_MEMORY` which uses `total_used_mb`. It returns average consumption for all hosts. This means when you see:
![nr_avg_total_memory](https://cloud.githubusercontent.com/assets/1812341/19679951/234e17d2-9aa4-11e6-8f37-95e30d0a1d12.jpg)

the API will return an average of three hosts, which is around `7.26` in that case.

Still, it's not "Get memory usage for the entire app" as NR claims, but this is sth of more value.